src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java

```diff
@@ -162,65 +164,123 @@ protected void doExecute(Task task, EntityResultRequest request, ActionListener<
+                                    Instant.ofEpochMilli(request.getEnd()),
+                                    executionStartTime,
+                                    Instant.now(),
+                                    null,
+                                    categoricalValues,
+                                    detector.getUser(),
+                                    indexUtil.getSchemaVersion(ADIndex.RESULT),
+                                    modelId
+                                )
                             )
                         );
                 }
             }
-            if (currentBulkRequest.numberOfActions() > 0) {
-                this.anomalyResultHandler.flush(currentBulkRequest, detectorId);
+
+            // split hot and cold entities
+            Pair<List<Entity>, List<Entity>> hotColdEntities = cache
+                .get()
+                .selectUpdateCandidate(cacheMissEntities.keySet(), detectorId, detector);
+
+            List<EntityFeatureRequest> hotEntityRequests = new ArrayList<>();
+            List<EntityFeatureRequest> coldEntityRequests = new ArrayList<>();
+
+            for (Entity hotEntity : hotColdEntities.getLeft()) {
+                double[] hotEntityValue = cacheMissEntities.get(hotEntity);
+                if (hotEntityValue == null) {
+                    LOG.error(new ParameterizedMessage("feature value should not be null: [{}]", hotEntity));
+                    continue;
+                }
+                hotEntityRequests
+                    .add(
+                        new EntityFeatureRequest(
+                            System.currentTimeMillis() + detector.getDetectorIntervalInMilliseconds(),
+                            detectorId,
+                            // will change once we know the anomaly grade. Set it to low since most of the entities should not count as
+                            // hot entities if the traffic is not totally random
+                            SegmentPriority.MEDIUM,
+                            hotEntity,
+                            hotEntityValue,
+                            request.getStart()
+                        )
+                    );
+            }
+
+            for (Entity coldEntity : hotColdEntities.getRight()) {
+                double[] coldEntityValue = cacheMissEntities.get(coldEntity);
+                if (coldEntityValue == null) {
+                    LOG.error(new ParameterizedMessage("feature value should not be null: [{}]", coldEntity));
+                    continue;
+                }
+                coldEntityRequests
+                    .add(
+                        new EntityFeatureRequest(
+                            System.currentTimeMillis() + detector.getDetectorIntervalInMilliseconds(),
+                            detectorId,
+                            // will change once we know the anomaly grade. Set it to low since most of the entities should not count as
```

Comment:
These are cold entities. Does this comment still apply?

Reply:
it does not.  Thanks for the catch. Fixed.